auth_enabled: false

server:
  log_level: warn
  http_listen_port: 3100

common:
  ring:
    instance_addr: 127.0.0.1
    kvstore:
      store: inmemory
  replication_factor: 1
  path_prefix: /tmp/loki

ingester:
  lifecycler:
    address: 127.0.0.1
    final_sleep: 0s
  # Any chunk not receiving new logs in this time will be flushed
  chunk_idle_period: 1h
  # All chunks will be flushed when they hit this age, default is 1h
  max_chunk_age: 1h
  # Loki will attempt to build chunks up to 1.5MB, flushing if chunk_idle_period or max_chunk_age is reached first
  chunk_target_size: 1048576
  wal:
    dir: {{ env "NOMAD_ALLOC_DIR" }}/data/wal
    flush_on_shutdown: true
    replay_memory_ceiling: "1G"

compactor:
  working_directory: {{ env "NOMAD_ALLOC_DIR" }}/data/tsdb-shipper-compactor
#  shared_store: filesystem

schema_config:
  configs:
    - from: 2024-07-11
      store: tsdb
      object_store: filesystem
      schema: v13
      index:
        prefix: tsdb_index_
        period: 24h

storage_config:
#  filesystem:
#    directory: /loki/chunks
  tsdb_shipper:
    active_index_directory: {{ env "NOMAD_ALLOC_DIR" }}/data/tsdb-shipper-active
    cache_location: {{ env "NOMAD_ALLOC_DIR" }}/data/tsdb-shipper-cache
    cache_ttl: 24h         # Can be increased for faster performance over longer query periods, uses more disk space
#    shared_store: filesystem

limits_config:
  reject_old_samples: true
  reject_old_samples_max_age: 168h

chunk_store_config:
#  max_look_back_period: 0s
  chunk_cache_config:
    embedded_cache:
      enabled: true
      max_size_mb: 1000
      ttl: 24h

query_range:
  parallelise_shardable_queries: false # helps with "context canceled" messages I'm getting
